internal/extproc/router/router.go

```diff
@@ -33,13 +33,12 @@ func New(config *filterapi.Config, newCustomFn x.NewCustomRouterFn) (x.Router, e
 // Calculate implements [x.Router.Calculate].
 func (r *router) Calculate(headers map[string]string) (backend *filterapi.Backend, err error) {
 	var rule *filterapi.RouteRule
-	for i := range r.rules {
-		_rule := &r.rules[i]
+	for _, _rule := range r.rules {
```

Comment:
This change saves us one line of getting the element by index.

Reply:
this `_rule := &r.rules[i]` was intentional to avoid copying the rule struct when ranging over

Reply:
ah I see. reverted